packages/server/lib/controllers/onboarding.controller.ts

```diff
@@ -161,12 +162,31 @@ class OnboardingController {
                 payload.progress = 3;
             }
 
-            const getRecords = await syncDataService.getAllDataRecords(connectionId, DEMO_GITHUB_CONFIG_KEY, environment.id, DEMO_MODEL);
-            if (!getRecords.success) {
+            let getRecords: Result<CustomerFacingDataRecord[], Error> = resultErr('failed_to_get_records');
+            const shouldReturnNewRecords = await featureFlags.isEnabled('new-records-return', 'global', false);
+            if (shouldReturnNewRecords) {
+                const { error, response: connection } = await connectionService.getConnection(connectionId, DEMO_GITHUB_CONFIG_KEY, environment.id);
```

Comment:
Connection is already checked in `checkIfConnectionExists` it returns the id

Reply:
fixed in 9e577400